### PR TITLE
https://trello.com/c/SVSoZiql

### PR DESCRIPTION
**Pull request by cschroed-usgs**
I made a few (hopefully) non-functional changes as a part of getting to know the code better.

### CODE REVIEW

File: src/main/webapp/js/utils/spatialUtils.js [hunk 1]
```diff
@@ -44,16 +53,28 @@ define([
...
+				deferred.resolveWith(context, [states]);
 			},
 			error: function () {
-				this.deferred.rejectWith(this.that, arguments);
+				deferred.rejectWith(context, arguments);
 			}
 		});
 
-		return deferred;
+		return deferred.promise();
```
Comment (isuftin) [inline src/main/webapp/js/utils/spatialUtils.js]: Have you been able to test these changes (switching to getting the promise from the deferred object)?

Comment (cschroed-usgs) [inline src/main/webapp/js/utils/spatialUtils.js]: Yeah. All automated tests still pass. The app behaves identically when I use it.

Comment (isuftin) [inline src/main/webapp/js/utils/spatialUtils.js]: What was the reason for adding the promise on the end? Just to finalize it, or...?

Comment (cschroed-usgs) [inline src/main/webapp/js/utils/spatialUtils.js]: A stack overflow answer gave the most succinct description of the difference that I've seen:
"A promise is something shared w/ other objects, while a deferred should be kept private."
http://stackoverflow.com/a/6824836

Explained another way, a promise is a read-only version of a deferred. We only want the function controlling the asynch operation to determine the resolution of the asynch task. If we return a deferred from the function, we give the caller of our function control to prematurely resolve or reject our asynch operation. If we return a promise from the function, the caller does not have control over how and when the asynch task reaches resolution -- they can only react to the resolution when it occurs.

Comment (isuftin) [inline src/main/webapp/js/utils/spatialUtils.js]: Fair enough. Sounds good to me! :D

Comment (cschroed-usgs) [inline src/main/webapp/js/utils/spatialUtils.js]: Thanks @isuftin. Is this mergerriffic yet?

**Comment by isuftin** [comment]
Sorry, thought you'd merged this ;)

**Comment by cschroed-usgs** [comment]
No worries. Thanks!

**Pull request merged**